packages/flutter_tools/lib/src/version.dart

```diff
@@ -171,6 +171,20 @@ class FlutterVersion {
     return branch;
   }
 
+  /// Returns true if `tentativeDescendantRevision` is a direct descendant to
+  /// the `tentativeAncestorRevision` revision on the Flutter framework repo
+  /// tree.
+  static bool checkRevisionAncestry({
```

Comment:
Why not make this an instance method that compares a revision that's passed to it against `_frameworkRevision`?

Reply:
If the flutter tool version is X, I'm going to almost always be comparing a version Y < X which is the version that created the present flutter project to a version Y < Z < X which is the threshold below which I want to perform some action to upgrade the current flutter project. i.e. X is not going to be used.

Reply:
Should still be an instance method so we can override it. Otherwise, we need to make sure all code paths that touch this are never called internally at Google.

In general this file should not have any static methods since it is already a singleton.

Reply:
Done